### ISSUE TITLE
Multitime scale Candles

### ISSUE THREAD

**Issue by JonastenHaaf** [freqtrade/freqtrade#6924]
Hello, I've got an open suggestion for an enhancement.

I think when backtesting a strategy or using hyperopt the statistical significance of the strategy depends very much on the numbers of trade done. In simple words, if a strategy does 10 excellent trades during backtest, this says nothing about its performance on the 11th test in live mode. If a bot has 5000 trades in backtest, and many of them are good, I can be confident about the statistical significance.

So, do you agree?
Do you have practical recommendations for increasing the number of trades?

I want to commend on the most practical ways that I am aware of, and conclude with a possible enhancement.
- If the candle time is low, say 1m, than 1y of data includes 525600 trade possibilities, so this issue is minimized. Yet, for larger candles it gets an issue.
- Of course, changing the strategy itself helps. Some of them have shorter trade durations.
- Using more Pairs to trade on helps.
- Including the number of trades explicitly in the hyper optimizer loss function helps.
- Using a different exchange, with more backtest data helps. Yet, the farer you go into the past, let's say 10 years, the less compareable the data gets.

I think about a possible enhancement.
If for example the candle time is 1h, and 5y backtest data, that leads to 43800 trade possibilies. That is not too much. Every day, that could be 00:00, 01:00, 02:00, 03:00, etc. Most likely the first timestamp depends on the time you enable the bot the first time.
What about backtesting the same strategy at different inter-candles, so at different starting points? So for example at 00:00, 01:00, 02:00, 03:00 and also at 00:10, 01:10, 02:10. Still, that are 1h candles, yet the number of data points is already doubled. And this idea scales. Using 1m candles for a 1h strategy improves the number of trade possibilites in backtest by factor of 60.
I think it will increase robustness. A 1h strategy that enters a position for example at 00:00 should also enter it at 00:01, as the relevant market conditions should change only slowly.

What do you think about it? What are the easiest ways of implementing it?

**Comment by xmatthias** [comment]
> Using a different exchange, with more backtest data helps

Be careful with this - pairs move differently on different exchanges - so you can't reliably transfer a strategy between exchanges without re-tuning.

---

Technically - you should be able to easily implement this within the strategy by using `1m` candles, and resampling them to use several different 1h candles (60 per candle?).

I don't see any changes to freqtrade for this necessary - everything (all resamplings) can be done within the strategy, and there you can also dynamically decide which column to use for your signal.

You can even make this changable with a "one line" change (a parameter defining the "Minute"/minutes) - allowing yourself to either use all 60 resamplings - or just 1 or 2 resamplings using a parameter.

---

If that's really a good approach is a different topic entirely. 
Your assumption is that a 1h candle from 00:00-01:00 shows the same/a very similar pattern than 00:01-01:01.

If we assume that's true - then you'd simply be replicating the signal from the 00:00 candle 60 times (so your mathematical changes are 60 times higher - but you're just running 60 times an (almost identical) candle - so you've not increased resiliancy - just given yourself the possibility to fit the strategy too much to past data.


Personally, i don't think that the candle/market behaves this way. If you observe markets, on many pairs, you'll have increased behaviour around the 59/01 minute - suggesting bots (or people?) running at 1h timeframe (obviously the same will repeat on other timeframes, too, at different levels so it's not limited to these minutes).
what this basically means is that a bot using 00:13-01:13 candles would see different patterns than a "regular hour" one. 
that's not per se wrong - but in my understanding you'd train/backtest the bot on 60x1h candles, but then run it on "regular" 1h candles - which would mean it'd be trained on potentially completely different patterns.

**Comment by xmatthias** [comment]
Closing this as this can be satisfied from within the strategy (given enough memory to use 1m candles).
